services/cd-service/pkg/service/overview.go

```diff
@@ -307,3 +320,22 @@ func transformSyncWindows(syncWindows []config.ArgoCdSyncWindow, appName string)
 	}
 	return envAppSyncWindows, nil
 }
+
+func extractPrNumber(sourceMessage string) *string {
```

Comment:
is the source message added by git? or is this from some script (thats written by us). Does it have pr number already.

Is this specific to github, or would all have it?

Also couldn't this be done with just the regex
```
	re := regexp.MustCompile("\\((#(([0-9])+))\\)")
	str := "something (#123)"
	res := re.FindAllStringSubmatch(str, 1)
	fmt.Println("Message :", res[0][2])
```

Reply:
The source message is added by git, yes. The PR number is present in the commit header for each commit.

Only the regex would not be enough because the commit header may contain '(', '(' or '#' in the message itself. There must be a filtering for those cases.

Reply:
yes but if there are multiple items of <left-paranthesis><hashtag>numbers<right-paranthesis>, do you do anything about it?

because thats the only thing the above regex looks for

Reply:
@mnishamk-freiheit the option you suggested has some problems. First of all, it does not work when there are multiple PR numbers for one commit. Additionally, it returns the first PR number that appears when in truth we want the last PR number.

Reply:
> yes but if there are multiple items of <left-paranthesis><hashtag>numbers<right-paranthesis>, do you do anything about it?
> 
> because thats the only thing the above regex looks for

Yes, the option that I inserted does that, precisely

Reply:
in above res is an array with all the pr numbers, res[0] is the first pr number.

Reply:
Your option works if the the PR numbers are within brackets for each one. For example, "Commit header (#PRNUMBER1) (#PRNUMBER2)". But can't the PR numbers be stored like "Commit header (#PRNUMBER1 #PRNUMBER2)". Because my implementation was done taking both cases into account.

Reply:
I'm still not sure if regex is the way to go or not, but since you'd used regex, figured could use regex for the whole thing. check with @hannesg  once

Reply:
Will do!

Reply:
@mnishamk-freiheit There is one case that my solution does not take into account. And after looking better the expression use for the regex, I think your approach is better and simpler